contents/blog/ab-testing-guide-for-engineers.md

```diff
@@ -0,0 +1,207 @@
+tags: 
+  - Guides
+  - Product metrics
+---
+
+A/B testing enables you to be confident that your product changes are having their intended effects. Rather than making decisions based on gut feeling or intuition, A/B testing provides empirical data to help you remove bias from decision making.
+
+Companies like [Netflix](https://uxdesign.cc/how-netflix-does-a-b-testing-87df9f9bf57c), [Uber](https://www.uber.com/en-GB/blog/supercharging-a-b-testing-at-uber/) and [Booking.com](https://booking.ai/tagged/ab-testing) use A/B testing to ship hundreds of changes a day. In this blog, we'll teach you basic concepts to get you started in A/B testing and supercharge your product growth!
+
+> **Note:** A/B tests are commonly referred to as "experiments".
+
+## How A/B testing works
+
+In A/B testing, you first decide on a goal metric and what change you'd like to see. For example, you may to increase conversion rate, or you may want to decrease customer churn rate. Then, once you've made your code changes and are ready to test, you **randomly** split your users into two (or more) groups:
+
+- The control group: these users will experience your product as is – without any of your new changes.
+- The test group: these users will experience your product with your new changes.
+
+You then run your A/B test for some time, usually a few days or weeks, to gather data and compare the differences in the goal metric between the different groups.
+
+![How A/B testing works](../images/blog/ab-testing-guide-for-engineers/ab-test-illustration.png)
+
+> **Note:** These groups are also referred to as "variants".
+
+## What makes a good A/B test?
+
+There are five parts to a good A/B test:
+
+### 1. Has a clear, measurable goal
+
+Like "increase conversion rate by 10%", or "decrease churn rate by 5%". Without a clear goal, it can be unclear what is a successful outcome of test.
+
+### 2. Has a clear hypothesis why your your changes will achieve your goal
+
+This focuses your testing on what changes you're making and why.
+
+Here's an example of a good hypothesis for a SaaS Product: "Adding a short tutorial video on the login page will increase user engagement by 30% because it will help new users understand how to use our product more effectively."
+
+> See our blog on [A/B testing mistakes](https://posthog.com/blog/ab-testing-mistakes#7-testing-an-unclear-hypothesis) to read more on how to avoid testing an unclear hypothesis.
+
+### 3. Tests as small of a change as *reasonably* possible 
+
+Ideally, an A/B test should change one thing at a time. This ensures that any change in user behavior can be attributed to the variable that was changed. If multiple variables are changed, it's difficult to determine which one caused the change in behavior.
+
+However, the caveat here is that too small of change can slow your team down. Since running A/B tests take time, if you're constantly testing small changes it may take long to ship large, meaningful changes.
```

Comment:
```suggestion
The caveat here is that too small of change can slow your team down. Since running A/B tests take time, if you're constantly testing small changes, it may take long to ship large, meaningful changes.
```